styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -554,6 +555,33 @@ private void setupMetrics(
               .count());
     });
 
+    final Function<String, Gauge<Long>> createResourceCountGauge = (resource) -> () -> {
+      try {
+        return storage.resource(resource).map(Resource::concurrency).orElse(0L);
+      } catch (IOException e) {
+        LOG.error("Failed to get resource {}", resource);
```

Comment:
We're repeating this pattern so much all over the place. Hard to imagine what will happen when configuration storage actually intermittently fails with IOExceptions - probably will have very ugly ripple effects.

What do you think of fronting storage with a CachingStorage, that does not throw, runs with the cached values forever if there are IOExceptions in contacting the real storage, and raises alarms if the failure persists for long? That's simpler code and improved operational understanding, via decoupling and modularization of concerns.